### ISSUE TITLE
Help needed installing formio

### ISSUE THREAD

**Issue by AllamSudhakara** [formio/formio#306]
Hi, I am new to node platforms. I tried installing based on the instructions provided. I first got issue with bcrypt which I changed to bcryptjs and npm install is running OK but it is not creating the app folder and no site is running at 3001 port. I also did not get prompts that were illustrated on GitHub. I am not quite sure what I did wrong. Could somebody provide me trouble shooting corners?

Regards,
Sudhakar

**Comment by zackurben** [comment]
Are you getting any errors? Is there an npm error file?

**Comment by AllamSudhakara** [comment]
I did not find error file generated yesterday. I used an installation of node.js and npm that I download prior. I do see a formio folder that got installed day before yesterday in node_modules folder but day before yesterday I had errors. Do I need to delete this folder and retry the installation again? If you have documentation on trouble shooting installation and in general step by step approach to installation, it would be very helpful. Does it matter if I cloned the solution to my profile instead of node.js folder? Please provide some clues on installation failure for me to successfully install the framework. Thanks for your help in advance.

**Comment by juanitomint** [comment]
ok first things first, which OS and versions are you runing in?
its probable better if you clone the project into its own folder so you wont mess up with nodejs install
you can ask for more detailed help in https://gitter.im/formio/formio
I'll be glad to help you

**Comment by zackurben** [comment]
thanks @juanitomint

**Comment by AllamSudhakara** [comment]
I have tried the installation on a fresh laptop with Windows Server 2016 standard evaluation edition. I am here by attaching the CMD output. It did exactly same as before on Windows 10. I had to change bcrypt to bcryptjs in Encrypt, package and index files. It runs with warnings but I do not get prompts and server does not run on port 3001. 
[FormIOCMDoutput.txt](https://github.com/formio/formio/files/870148/FormIOCMDoutput.txt). Help is highly appreciated. I have latest node installed.

**Comment by juanitomint** [comment]
ok windows it's not the best platform for server apps, in fact it's complaining about file system
Are you able to setup virtual machines? ie: install virtualbox and vagrant ?

**Comment by helderfsanto** [comment]
What do you mean by saying it's not the best platform for server apps. Are
you saying it's better to use another os?

A Sáb, 25/03/2017, 22:36, Juan Ignacio Borda <notifications@github.com>
escreveu:

**Comment by zackurben** [comment]
@AllamSudhakara @helderfsanto it should be fine to run the server on *nix or win servers, however it may take more kicking on windows..

That being said, it seems like you replaced bcrypt with bcryptjs; if you just changed the dependency than that wont work, they aren't backwards compatible, so unless you changed code, you possibly made a larger issue. Bcrypt uses native dependencies included with MS Visual Studios, so you will need that.

You can get all the ms build dependencies with `npm install --global --production windows-build-tools`

Then `npm install --msvs_version=2015`

Note: you may need to change the msvs, depending on what version of visual studios you have, additionally you will need python 2.7 for node-gyp to work correctly.

**Comment by AllamSudhakara** [comment]
Thank you Zack. I am very fascinated FormIO kind of frame works. I will revert back the changes and build dependencies and check out. I do remember issuing msvs_version 2015 before after reading from one of the post. I will check out on a clean computer that I have. Hopefully it works.

**Comment by AllamSudhakara** [comment]
I am getting further errors in trying to install windows build tools and msvs version. I am attaching the output of the command file. Is full reinstall ideal as the current nodejs, npm and formio projects might have been not in sync at many places?
[Windows10InstallProblems.txt](https://github.com/formio/formio/files/877264/Windows10InstallProblems.txt)

**Comment by sonicos** [comment]
if you open a command prompt and type 
```
powershell
```
and hit enter, what response do you get?

**Comment by dbaleeds** [comment]
I have issues as well trying to install on Windows 7 / node version v6.10.3.

Same bcrypt error, I have installed all MS build dependancies but this does not resolve.

**Comment by travist** [comment]
This happens when it cannot install because of node-gyp issues. The resolution for this is to install the windows build tools.

```
npm install --global --production windows-build-tools
```

Once you do that, verify that it installs everything without error. You will then need to delete your ```node_modules``` folder within this repo, and then type ```npm install``` again. Make sure you do not see any major errors involving node-gyp.

Hope this helps,

Travis.

**Comment by dbaleeds** [comment]
I have tried this with a fresh install and with deleting the NPM modules folder and I am still having an issue. However this time I am getting a different error message: 

'Error: Cannot find module 'mongoose''

**Comment by dbaleeds** [comment]
Hi Travis,

Thanks your help with this!

I have tried to install the windows build tools but now I am still receiving an error. The error is now different.

I have attached the error log to the email, I have also add the comment into the GitHub issues in case someone else in your team can help,

Again thanks for your help with this,

Thanks, David.

**Comment by juanitomint** [comment]
hey! @dbaleeds  can you post the result of command ?
`npm i `

**Comment by dbaleeds** [comment]
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated to-iso-string@0.0.2: to-iso-string has been deprecated, use @segment/to-iso-string instead.
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue

> pre-commit@1.2.2 install C:\inetpub\formio-master\node_modules\pre-commit
> node install.js


> bcrypt@0.8.7 install C:\inetpub\formio-master\node_modules\bcrypt
> node-gyp rebuild


C:\inetpub\formio-master\node_modules\bcrypt>if not defined npm_config_node_gyp (node "C:\Users\user12246\AppData\Roaming\npm\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-
gyp\bin\node-gyp.js" rebuild )  else (node "" rebuild )
gyp ERR! build error
gyp ERR! stack Error: Command failed: reg query "HKLM\Software\Microsoft\MSBuild\ToolsVersions" /s /reg:32
gyp ERR! stack ERROR: Registry editing has been disabled by your administrator.
gyp ERR! stack
gyp ERR! stack Can't find "msbuild.exe". Do you have Microsoft Visual Studio C++ 2008+ installed?
gyp ERR! stack     at C:\Users\user12246\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\build.js:141:25
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:211:5)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at maybeClose (internal/child_process.js:886:16)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\user12246\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\inetpub\formio-master\node_modules\bcrypt
gyp ERR! node -v v6.10.3
gyp ERR! node-gyp -v v3.6.1
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! bcrypt@0.8.7 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the bcrypt@0.8.7 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\user12246\AppData\Roaming\npm-cache\_logs\2017-06-05T13_23_20_992Z-debug.log

**Comment by denisekay** [comment]
We have support packages available if you would like help with this from our dev team.  Feel free to email me at denise@form.io if you are interested

**Comment by sonicos** [comment]
The error reads that you don't have permission to edit the registry. You
need to run the install in the context of an administrator so that you can
install all the build tools you need

**Comment by dbaleeds** [comment]
Thanks, this is my first thought however it didnt fix, I got the same error. 

It looks like there are issues with 'node-gyp'. I have searched around and it seems a common problem in windows but I have not been able to resolve.

**Comment by dbaleeds** [comment]
I finally got this working after many lost hours! I hope this fix will help anyone out if they come across a similier situation. 

Long story short 'bcrypt' does not play will with windows - this is a problem still not resolved at Microsoft.  I swapped to 'bcrypt-nodejs' the process is fairly straightforward; 

change package.json to use 'bcrypt-nodejs' instead.

npm install --save bcrypt-nodejs && npm uninstall --save bcrypt

Swap instances of require('bcrypt'); to require('bcrypt-nodejs');

Then change the hash function in encypt.js from/to;
bcrypt.hash(text, salt, function(error, hash) {
bcrypt.hash(text, salt, null, function(error, hash) {

Hope this helps anyone wishing to install on windows..
